evaluationStage.go

```diff
@@ -419,15 +434,45 @@ func separatorStage(left interface{}, right interface{}, parameters Parameters)
 }
 
 func inStage(left interface{}, right interface{}, parameters Parameters) (interface{}, error) {
-
-	for _, value := range right.([]interface{}) {
-		if left == value {
+	v := reflect.ValueOf(right)
+	for i := 0; i < v.Len(); i++ {
+		value := reflect.Indirect(v.Index(i)).Interface()
+		if valuesEqual(left, value) {
 			return true, nil
 		}
 	}
 	return false, nil
 }
 
+func coerceFloat(v reflect.Value) float64 {
+	switch kind := v.Kind(); kind {
+	case reflect.Float32, reflect.Float64:
+		return v.Float()
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return float64(v.Uint())
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return float64(v.Int())
```

Comment:
Do we run the risk of overflows causing false positives/negatives when an int64 / uint64 value is nearing is upper bounds?

Reply:
There is some risk here, but not of overflow. Rather, we risk losing information for numbers with a magnitude greater than 53 bits, possibly leading to bad comparisons.

I'd consider this risk to be quite small, and worth it for the sake of simplicity.